File: get_backend_port.py
#!/usr/bin/env python3
"""Get the current backend port from running processes."""

import subprocess
import re
import sys

def get_backend_port():
    """Get the port of the running backend server."""
    try:
        # Get all uvicorn processes
        result = subprocess.run(['ps', 'aux'], capture_output=True, text=True)
        lines = result.stdout.split('\n')
        
        for line in lines:
            if 'uvicorn' in line and 'app.main:app' in line:
                # Extract port from the command line
                port_match = re.search(r'--port\s+(\d+)', line)
                if port_match:
                    return int(port_match.group(1))
        
        # If no port found, return default
        return 8000
    except Exception as e:
        print(f"Error getting backend port: {e}", file=sys.stderr)
        return 8000

if __name__ == "__main__":
    port = get_backend_port()
    print(port)

File: frontend/scripts/get-backend-port.js
#!/usr/bin/env node
/**
 * Get the current backend port from running processes
 */

const { exec } = require('child_process');
const { promisify } = require('util');

const execAsync = promisify(exec);

/**
 * Get the port of the running backend server
 * @returns {Promise<number>} Backend port number
 */
async function getBackendPort() {
  try {
    // Get all processes
    const { stdout } = await execAsync('ps aux');
    const lines = stdout.split('\n');
    
    for (const line of lines) {
      if (line.includes('uvicorn') && line.includes('app.main:app')) {
        // Extract port from command line
        const portMatch = line.match(/--port\s+(\d+)/);
        if (portMatch) {
          return parseInt(portMatch[1]);
        }
      }
    }
    
    // If no port found, return default
    return 8000;
  } catch (error) {
    console.error('Error getting backend port:', error);
    return 8000;
  }
}

/**
 * Main function
 */
async function main() {
  const port = await getBackendPort();
  console.log(port);
}

// Run if called directly
if (require.main === module) {
  main();
}

module.exports = { getBackendPort };
